include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -392,7 +392,7 @@ namespace internal
                   quad[my_q][q].get_tensor_basis()[0], update_default);
             }
 
-          face_data[my_q].descriptor.resize(n_hp_quads);
+          face_data[my_q].descriptor.reserve(n_hp_quads * 2);
```

Comment:
@kronbichler I am wondering if we couldn't write here:
```suggestion
          face_data[my_q].descriptor.resize(n_hp_quads * std::max<unsigned int>(1, dim - 1));
```
with only the relevant descriptors filled.

Given a `hp_quad_index` and a `face_type`, we could access the correct face-quadrature in `FEEvaluationBase` via:
`face_data[quad_no].descriptor[std::max<unsigned int>(1, dim - 1) * hp_quad_index + face_type]` without the need of a complicated logic. `face_type` could  be e.g. stored in `FaceToCellTopology`. In 3D, `face_type` could be 0 (tet) or 1 (quad). I am currently trying to propagate the relevant information into the setup routines.

Reply:
This looks reasonable. Please go ahead with the simplified version you suggest here.

Reply:
I am on it. If everything goes according to plans we should be able to evaluate face integrals for wedges today.